examples/svm/plot_svm_kernels.py

```diff
@@ -1,93 +1,272 @@
+#
+# Polynomial kernel
+# *****************
+# The polynomial kernel changes the notion of similarity. The kernel function
+# is defined as:
+#
+# .. math::
+#   K(\mathbf{x}_1, \mathbf{x}_2) = (\gamma \cdot \
+#       \mathbf{x}_1^\top\mathbf{x}_2 + r)^d
+#
+# where :math:`{d}` is the degree (`degree`) of the polynomial, :math:`{\gamma}`
+# (`gamma`) controls the influence of each individual training sample on the
+# decision boundary and :math:`{r}` is the bias term (`coef0`) that shifts the
+# data up or down. Here, we use the default value for the degree of the
+# polynomal in the kernel funcion (`degree=3`). When `coef0=0` (the default),
+# the data is only transformed, but no additional dimension is added. Using a
+# polynomial kernel is equivalent to creating
+# :class:`~sklearn.preprocessing.PolynomialFeatures` and then fitting a
+# :class:`~sklearn.svm.SVC` with a linear kernel on the transformed data,
+# although this alternative approach would be computationally expensive for most
+# datasets.
+plot_training_data_with_decision_boundary("poly")
+
+# %%
+# The polynomial kernel with the kernel coefficient `gamma=2` bends around
+# the training data well. The margins to both sides of the hyperplane bend with
+# it.
+#
+# RBF kernel
+# **********
+# The radial basis function (RBF) kernel, also known as the Gaussian kernel, is
+# the default kernel for Support Vector Machines in scikit-learn. It measures
+# similarity between two data points in infinite dimensions and then approaches
+# classification by majority vote. The kernel function is defined as:
+#
+# .. math::
+#   K(\mathbf{x}_1, \mathbf{x}_2) = \exp\left(-\gamma \cdot
+#       {\|\mathbf{x}_1 - \mathbf{x}_2\|^2}\right)
+#
+# where :math:`{\gamma}` (`gamma`) controls the influence of each individual
+# training sample on the decision boundary.
+#
+# The larger the euclidean distance between two points
+# :math:`\|\mathbf{x}_1 - \mathbf{x}_2\|^2`
+# the closer the kernel function is to zero. This means that two points far away
+# are more likely to be dissimilar.
+plot_training_data_with_decision_boundary("rbf")
 
-    plt.xlim(x_min, x_max)
-    plt.ylim(y_min, y_max)
+# %%
+# In the plot we can see how the decision boundaries tend to contract around
+# data points, that are close to each other.
+#
+# Sigmoid kernel
+# **************
+# The sigmoid kernel function is defined as:
+#
+# .. math::
+#   K(\mathbf{x}_1, \mathbf{x}_2) = \tanh(\gamma \cdot
+#       \mathbf{x}_1^\top\mathbf{x}_2 + r)
+#
+# where :math:`{\gamma}` (`gamma`) controls the influence of each individual
+# training sample on the decision boundary and :math:`{r}` is the bias term
+# (`coef0`) that shifts the data up or down.
+#
+# In the sigmoid kernel, the similarity between two data points is computed
+# using the hyperbolic tangent function (tanh). The kernel function takes into
+# account the dot product of the two points (x1 and x2), scaled by the kernel
+# coefficient γ, and shifted by a bias term r.
```

Comment:
I suggest merging these two paragraphs into a single one to avoid redundancies:

```suggestion
# where the similarity between two data points is computed using the hyperbolic
# tangent function (:math:`tanh`). The kernel function takes into account the
# dot product of the two points (:math:`\mathbf{x}_1` and :math:`\mathbf{x}_2`),
# scaled by the kernel coefficient :math:`{\gamma}`, and shifted by a bias term
# :math:`{r}` (`coef0`).
```

Reply:
I'd rather leave the paragraph explaining the components of the mathematical formula separate, directly following the formula like in the other paragraphs, if this is alright. But I have improved the paragraph with :math: notation. :)